### DISCUSSION TITLE
Connect via Nginx reverse proxy show status in Logdy "Not connected"

### DISCUSSION THREAD

**Discussion by krot4u** [logdyhq/logdy-core#21] [Q&A] [answered]
This my systemd unit:
[Unit]
Description=XLX Logs via Browser

[Service]
Type=simple
Restart=on-failure
RestartSec=3
User=root
Group=root
WorkingDirectory=/var/log/
ExecStart=/bin/bash -c tail -f /var/log/xlxd.log | logdy --port 1234 --ui-ip 0.0.0.0 --ui-pass PassWord --no-analytics --no-updates
StandardOutput=/var/log/logdy.log
StandardError=/var/log/logdy.log
SyslogIdentifier=logdy
KillMode=process

[Install]
WantedBy=multi-user.target

systemctl start logdy - start and running fine.

Direct connect to host on port 1234 is ok - Logdy working as expected.
Now I have Nginx Proxy Manager running on another host.
And there is Proxy Host rule is configured as folow:
![image](https://github.com/logdyhq/logdy-core/assets/6033525/afc77fa9-7d31-434f-84d9-f6b9f27b3231)

Now when I open this URL on nginx proxy I get connected to Logdy interface, but it not streaming the log file...
![image](https://github.com/logdyhq/logdy-core/assets/6033525/8f1ea88d-14b1-4fe0-9930-b8f08f94d4cf)

Where did i made mistake?

**Comment by krot4u** [comment]
Hi! @PeterOsinski Any thoughts on this?

**Reply by PeterOsinski** [reply]
Hi @krot4u sorry for later reply, I didnt see that message before. I can see from the screenshot that the connection with the process is not established through websockets. Logdy serves UI over HTTP server but the messages are streamed over websockets. So the most probable cause of this is because the proxy (nginx) is not forwarding websocket traffic (https://nginx.org/en/docs/http/websocket.html). I think you can test it by trying to connect to the websocket in postman (switch the protocol to websocket and connect to hostname:port/ws?should_follow=true).
Let me know if it helped!

**Reply by krot4u** [reply]
I guess the problem is that WS is not TLS enabled.
Here is the error from Chrome console:
index-8qFs_H9l.js:7117 Mixed Content: The page at 'https://x.online/' was loaded over HTTPS, but attempted to connect to the insecure WebSocket endpoint 'ws://x.online/ws?password=SecretPass'. This request has been blocked; this endpoint must be available over WSS.

**Reply by PeterOsinski** [reply]
Good point, currently the protocol is hardcoded in the UI
https://github.com/logdyhq/logdy-ui/blob/6fe54fde15b18e075542e85f60d4133948cbdaa3/src/App.vue#L358
```js
const socket = new WebSocket('ws://' + window.location.host + '/ws?password=' + store.getPassword());
``` 
I'll change it to `window.location.protocol` so that if you load the UI through proxy with https, UI should try to connect to `wss`

**Reply by christianarrer** [reply]
You could use a subfilter as temp solution like this:

sub_filter_types text/xml text/css text/javascript;
sub_filter 'new WebSocket("ws://"+window.location.host+"/' 'new WebSocket("wss://"+window.location.host+"/';
sub_filter_once off;

**Reply by christianarrer** [reply, marked as answer]
The full nginx locaton section of a virtual host with ssl enabled works for me like this:

```
location / {
          auth_basic "Stable Diffusion Log";
          auth_basic_user_file c:\\users\\sd\\.htpasswd;
          proxy_pass http://192.168.42.2:8081/;
          sub_filter_types text/xml text/css text/javascript;
          sub_filter 'new WebSocket("ws://"+window.location.host+"/' 'new WebSocket("wss://"+window.location.host+"/';
          sub_filter_once off;
          proxy_buffering off;
          proxy_http_version 1.1;
          proxy_set_header Upgrade $http_upgrade;
          proxy_set_header Connection "upgrade";
          proxy_set_header Host $host;
          proxy_set_header X-Forwarded-Host $host;
          proxy_set_header X-Forwarded-Proto $scheme;
}
```

The ability of starting logdy with a subpath like ```http://192.168.42.2:8081/<some dir>``` would be a nice feature.

**Reply by krot4u** [reply]
Great advice! Thanks! It's working like a charm!

**Reply by pestevao** [reply]
> The full nginx locaton section of a virtual host with ssl enabled works for me like this:
> 
> ```
> location / {
>           auth_basic "Stable Diffusion Log";
>           auth_basic_user_file c:\\users\\sd\\.htpasswd;
>           proxy_pass http://192.168.42.2:8081/;
>           sub_filter_types text/xml text/css text/javascript;
>           sub_filter 'new WebSocket("ws://"+window.location.host+"/' 'new WebSocket("wss://"+window.location.host+"/';
>           sub_filter_once off;
>           proxy_buffering off;
>           proxy_http_version 1.1;
>           proxy_set_header Upgrade $http_upgrade;
>           proxy_set_header Connection "upgrade";
>           proxy_set_header Host $host;
>           proxy_set_header X-Forwarded-Host $host;
>           proxy_set_header X-Forwarded-Proto $scheme;
> }
> ```
> 
> The ability of starting logdy with a subpath like `http://192.168.42.2:8081/<some dir>` would be a nice feature.

Thank you! 
This should be in the official logdy documentation!
